Automattic/WP-Job-Manager
# Allows for the option (filter) for shortcodes to be stripped

Fixes #1146

I chose to strip the shortcodes earlier and check the post type instead of removing `do_shortcode`. Hopefully this would handle the case where the theme calls `get_content` outside of a job manager template.

Note: I'm not sure this should actually be merged. This still requires a snippet but could be completely replaced with another snippet added to our snippet library:
```php
add_filter( 'the_content', function( $content ) {
	if ( 'job_listing' !== get_post_type() ) {
		return $content;
	}
	return strip_shortcodes( $content );
} );
```

#### Changes proposed in this Pull Request:

* Adds option to strip shortcodes from content if `job_manager_strip_job_shortcodes` returns true.

#### Testing instructions:

* Add job listing with a shortcode.
* Verify the shortcode appears when viewing it on the frontend.
* Add snippet:
`add_filter( 'job_manager_strip_job_shortcodes', '__return_true' );`
* Refresh job listing and verify the shortcode doesn't appear.
